pkg/cmd/experimental/diagnostics/diagnostics.go

```diff
@@ -0,0 +1,214 @@
+client, master, and node, and if found, use them for troubleshooting
+those components. If master/node config files are not found, the tool
+assumes they are not present and does diagnostics only as a client.
+
+You may also specify config files explicitly with flags below, in which
+case you will receive an error if they are not found or invalid.
+
+    $ %[1]s --master-config=/etc/openshift/master.yaml
+
+Subcommands may be used to scope the troubleshooting to a particular
+component and are not limited to using config files; you can and should
+use the same flags that are actually set on the command line for that
+component to configure the diagnostic.
+
+    $ %[1]s node --master=https://master.openshift.example.com:8443 --cert-dir=...
+
+NOTE: This is an alpha version of diagnostics and will change significantly.
+NOTE: Global flags (from the 'options' subcommand) are ignored here but
+can be used with subcommands.
+`
+
+func NewCommand(name string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "diagnostics",
+		Short: "This utility helps you understand and troubleshoot OpenShift v3.",
+		Long:  fmt.Sprintf(longAllDescription, name),
+	}
+	diagFlags := types.NewFlags(cmd.PersistentFlags())
+	addFlags(cmd, diagFlags)
+	cmd.Flags().StringVar(&diagFlags.ClientConfigPath, config.OpenShiftConfigFlagName, "", "Path to the client config file.")
+	cmd.Flags().StringVar(&diagFlags.MasterConfigPath, "master-config", "", "Path to the master config file.")
+	cmd.Flags().StringVar(&diagFlags.NodeConfigPath, "node-config", "", "Path to the node config file.")
+
+	/* There is some weirdness with diagnostics flag usage. The same flags
+	   object is shared between the (implied) "all" cmd and the client cmd.
+
+	   We actually use the client factory built in the "client" subcommand for
+	   discovery in either case, and that adds flags to the "client" cmd which
+	   need a target for putting results in; we do not want the factory to add
+	   those flags to the "all" command (the only client option there is a
+	   config file), but those flags have to exist somewhere for the factory
+	   to set values and look them up later (even though on the "all" command
+	   the flags will not exist).
+
+	   So the flags object from client cmd is reused for the "all" command.
+	*/
+
+	ccmd, factory := NewClientCommand(name+" client", diagFlags)
+	cmd.AddCommand(ccmd)
+	cmd.AddCommand(NewMasterCommand(name + " master"))
+	cmd.AddCommand(NewNodeCommand(name+" node", diagFlags))
+	cmd.AddCommand(NewOptionsCommand())
+
+	cmd.Run = func(c *cobra.Command, args []string) {
+		runInit(c, diagFlags)
+		diagFlags.CanCheck[types.ClientTarget] = true
+		diagFlags.CanCheck[types.MasterTarget] = true
+		diagFlags.CanCheck[types.NodeTarget] = true
+		run.Diagnose(diagFlags, factory)
+	}
+	return cmd
+}
+
+// Used in the command definition for common flags that we want to be visible in usage
+func addFlags(cmd *cobra.Command, flags *types.Flags) {
```

Comment:
Move the bind method closer to the type being bound and use an object to allow overriding of the flags.  That leaves it easy to bind the defaults for consistency, but possible to re-use the flags with slightly different names.  See https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/client/clientcmd/overrides.go as an example.